Extend/OrderController.php

```diff
@@ -0,0 +1,249 @@
+use \Wirecard\Oxid\Core\Helper;
+use \Wirecard\Oxid\Extend\Order;
+
+use \Psr\Log\LoggerInterface;
+
+/**
+ * Class Order
+ *
+ * @mixin \OxidEsales\Eshop\Application\Controller\OrderController
+ */
+class OrderController extends OrderController_parent
+{
+    /**
+     * Extends the parent init function and finalizes the order in case it was a Wirecard payment method
+     */
+    public function init()
+    {
+        parent::init();
+
+        $oConfig = Registry::getConfig();
+        $sWdPaymentRedirect = $oConfig->getRequestParameter('wdpayment');
+        $oSession = Registry::getSession();
+        $sWdSessionToken = $oSession->getVariable('wdtoken');
+
+        if ($this->_isPaymentFinished($sWdSessionToken, $sWdPaymentRedirect)) {
+            $sShopBaseUrl = $oConfig->getShopUrl();
+            $sLanguageCode = Registry::getLang()->getBaseLanguage();
+
+            $aParams = array(
+                'lang' => $sLanguageCode,
+                'force_sid' => $oSession->getId(),
+                'stoken' => $oSession->getSessionChallengeToken(),
+                'actcontrol' => 'order',
+                'cl' => 'order',
+                'fnc' => 'execute',
+                'sDeliveryAddressMD5' => $this->getDeliveryAddressMD5(),
+                'challenge' => '',
+                'ord_agb' => '1',
+                'oxdownloadableproductsagreement' => '0',
+                'oxserviceproductsagreement' => '0',
+                'wdtoken' => $sWdSessionToken
+            );
+
+            $sParamStr = http_build_query($aParams);
+
+            $sNewUrl = $sShopBaseUrl . 'index.php?' . $sParamStr;
+            header('Location: '.$sNewUrl);
+        }
+    }
+
+    /**
+     * Redirects to the PaymentSDK if the payment is a Wirecard method
+     *
+     * @inheritdoc
+     *
+     * @return void
+     */
+    public function execute()
+    {
+        $oBasket = $this->getSession()->getBasket();
+        $oUser = $this->getUser();
+
+        if ($this->_isHandledByOxid($oBasket)
+            || !$this->_shouldCreateOrder($oBasket, $oUser)) {
+            return;
+        }
+
+        if ($oBasket->getProductsCount()) {
+            $oOrder = $this->_createOrder($oBasket, $oUser);
+
+            $this->_handleTransaction($oBasket, $oOrder);
+        }
+    }
+
+    /**
+     * Checks if it can be handled by oxid
+     *
+     * @param Basket $oBasket
+     *
+     * @return bool
+     */
+    private function _isHandledByOxid($oBasket)
+    {
+        $oConfig = Registry::getConfig();
+        $sWdPaymentRedirect = $oConfig->getRequestParameter('wdtoken');
+        $sWdSessionToken = $this->getSession()->getVariable('wdtoken');
+
+        if (!Helper::isModulePaymentMethod($oBasket->getPaymentId())
+            || $this->_isPaymentFinished($sWdSessionToken, $sWdPaymentRedirect)) {
+            parent::execute();
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Handles Wirecard payment transaction
```

Comment:
Shouldn't we avoid Wirecard? Could use "Custom payment" or "Module payment" instead. Also in other descriptions.